File: fib.py
def fib(n):
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)

def main():
    n = 40
    ans = fib(n)
    print(f'fib({n}) = {ans}')

if __name__ == '__main__':
    main()

File: examples/fib.rs
use interpreter::{Val, Vm, make_fib};

fn main() {
    let mut vm = Vm::new();
    let fib = make_fib();
    let n = 40;
    let ans = vm.run_with_func(fib, [Val::Int(n)]).unwrap();
    println!("fib({n}) = {ans:?}");
}
